lib/amazon/event_catcher_stream.rb

```diff
@@ -0,0 +1,184 @@
+#
+# Uses the AWS Config or CloudWatch service to monitor for events.
+#
+# AWS Config or CloudWatch events are collected in an SNS Topic.  Each appliance uses a unique
+# SQS queue subscribed to the AWS Config topic.  If the appliance-specific queue
+# doesn't exist, this event monitor will create the queue and subscribe the
+# queue to the AWS Config topic.
+
+#
+class Amazon::EventCatcherStream
+  class ProviderUnreachable < StandardError
+  end
+
+  #
+  # Creates an event monitor
+  #
+  # @param [Amazon::EventCatcher] event_catcher
+
+  def initialize(event_catcher)
+    @event_catcher = event_catcher
```

Comment:
Not a fan of passing in the whole event_catcher worker instance, feels lazy as opposed to passing in just what is needed.

Reply:
right, though we use connection methods from the catcher, so I can't pass those. The event_catcher object acts more like a client, maybe I should separate that?

Reply:
Although, the connection can be persistent. So maybe I can pass that? Not sure why we made new connection for every call (this is from MIQ)

Reply:
actually all of this can be in 1 class, so lets just simplify it this way :-)

Reply:
:+1: